substrate/frame/support/procedural/src/pallet/expand/tt_default_parts.rs

```diff
@@ -135,7 +135,51 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 		.any(|c| matches!(c.composite_keyword, CompositeKeyword::SlashReason(_)))
 		.then_some(quote::quote!(+ SlashReason));
 
+	let is_part_defined_call = generate_is_part_defined(
+		def.call.is_some(),
+		"__substrate_call_check",
+		"#[pallet::call]",
+		"is_call_part_defined",
+	);
+	let is_part_defined_event = generate_is_part_defined(
+		def.event.is_some(),
+		"__substrate_event_check",
+		"#[pallet::event]",
+		"is_event_part_defined",
+	);
+	let is_part_defined_inherent = generate_is_part_defined(
+		def.inherent.is_some(),
+		"__substrate_inherent_check",
+		"#[pallet::inherent]",
+		"is_inherent_part_defined",
+	);
+	let is_part_defined_origin = generate_is_part_defined(
+		def.origin.is_some(),
+		"__substrate_origin_check",
+		"#[pallet::origin]",
+		"is_origin_part_defined",
+	);
+	let is_part_defined_validate_unsigned = generate_is_part_defined(
+		def.validate_unsigned.is_some(),
+		"__substrate_validate_unsigned_check",
+		"#[pallet::validate_unsigned]",
+		"is_validate_unsigned_part_defined",
+	);
+	let is_part_defined_genesis_config = generate_is_part_defined(
+		def.genesis_config.is_some(),
+		"__substrate_genesis_config_check",
+		"#[pallet::genesis_config]",
+		"is_genesis_config_defined",
+	);
+
 	quote::quote!(
+		#is_part_defined_call
+		#is_part_defined_event
+		#is_part_defined_inherent
+		#is_part_defined_origin
+		#is_part_defined_validate_unsigned
+		#is_part_defined_genesis_config
```

Comment:
here are all the helper macro generated. But honeslty as nobody should write the part manually we could also remove all those checks (except is_std_enabled_genesis_config)

Reply:
We currently have the ability to split pallet definitions over multiple files, dont know if that changes the need for this.

Reply:
I think it shouldn't relate. They should only be used by construct_runtime and runtime macro to assert the consistency of the pallet parts.

The thing is we allow to manually give the pallet parts and at the same time we allow to generate them automatically. So this `is_part_defined` is a bit redundant with the code for automatic generation of pallet parts.